### ISSUE TITLE
The killall command leaves autofs in broken state on EL9

### ISSUE THREAD

**Issue by jblomer** [cvmfs/cvmfs#3174]
After `cvmfs_config killall` it seems that the final `systemctl reload autofs` triggers a bug in the automounter so that it quits with a SEGFAULT. Investigate and submit bug report to RH.

**Comment by vvolkl** [comment]
Sample of the error in the system log:
```
Jul 26 14:39:37 cvm-perf13 systemd[1]: Reloaded Automounts filesystems on demand.
Jul 26 14:39:37 cvm-perf13 kernel: automount[1531820]: segfault at 7f5c5fea8510 ip 00007f5c5fea8510 sp 00007f5c5d1f9c98 error 14 in libnss_sss.so.2[7f5c5fead000+2000] likely on CPU 13 (core 25, socket 0)
Jul 26 14:39:37 cvm-perf13 kernel: Code: Unable to access opcode bytes at 0x7f5c5fea84e6.
Jul 26 14:39:37 cvm-perf13 systemd[1]: Started Process Core Dump (PID 1531824/UID 0).
Jul 26 14:44:37 cvm-perf13 systemd[1]: systemd-coredump@169-1531824-0.service: Service reached runtime time limit. Stopping.
Jul 26 14:44:37 cvm-perf13 systemd[1]: systemd-coredump@169-1531824-0.service: Failed with result 'timeout'.
Jul 26 14:44:37 cvm-perf13 systemd[1]: autofs.service: Main process exited, code=dumped, status=11/SEGV
Jul 26 14:44:37 cvm-perf13 systemd[1]: autofs.service: Failed with result 'core-dump'.
Jul 26 14:45:43 cvm-perf13 systemd[1]: Starting Automounts filesystems on demand...
Jul 26 14:45:43 cvm-perf13 systemd[1]: tmp-autowBh4Jc.mount: Deactivated successfully.
Jul 26 14:45:43 cvm-perf13 systemd[1]: Started Automounts filesystems on demand.
Jul 26 14:45:44 cvm-perf13 systemd[1]: tmp-autoYBMnPr.mount: Deactivated successfully.
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531885]: (cvmfs-config.cern.ch) looks like cvmfs has been crashed previously
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531909]: re-building cache database
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531885]: (cvmfs-config.cern.ch) switching proxy from (none) to DIRECT (set proxies)
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531885]: (cvmfs-config.cern.ch) geographic order of servers retrieved from cvmfs-s1fnal.opensciencegrid.org
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531885]: (cvmfs-config.cern.ch) switching proxy from (none) to DIRECT (cloned)
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531885]: (cvmfs-config.cern.ch) CernVM-FS: linking /cvmfs/cvmfs-config.cern.ch to repository cvmfs-config.cern.ch
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531941]: (lhcb.cern.ch) looks like cvmfs has been crashed previously
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531941]: (lhcb.cern.ch) switching proxy from (none) to http://188.184.98.18:3128 (set proxies)
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531941]: (lhcb.cern.ch) geographic order of servers retrieved from cvmfs-stratum-one.ihep.ac.cn
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531941]: (lhcb.cern.ch) switching proxy from http://188.184.98.18:3128 to http://137.138.77.124:3128 (geosort)
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531941]: (lhcb.cern.ch) switching proxy from (none) to http://137.138.148.214:3128 (cloned)
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531941]: (lhcb.cern.ch) switching proxy from http://137.138.148.214:3128 to DIRECT (set proxies)
Jul 26 14:45:45 cvm-perf13 cvmfs2[1531941]: (lhcb.cern.ch) CernVM-FS: linking /cvmfs/lhcb.cern.ch to repository lhcb.cern.ch
Jul 26 14:46:12 cvm-perf13 systemd[1]: Started /usr/bin/systemctl start man-db-cache-update.
Jul 26 14:46:12 cvm-perf13 systemd[1]: Starting man-db-cache-update.service...
Jul 26 14:46:12 cvm-perf13 systemd[1]: Reloading.
Jul 26 14:46:12 cvm-perf13 systemd-rc-local-generator[1532122]: /etc/rc.d/rc.local is not marked executable, skipping.
Jul 26 14:46:12 cvm-perf13 systemd[1]: Queuing reload/restart jobs for marked units…
Jul 26 14:46:12 cvm-perf13 systemd[1]: Stopping Automounts filesystems on demand...
Jul 26 14:46:12 cvm-perf13 systemd[1]: cvmfs-cvmfs\x2dconfig.cern.ch.mount: Deactivated successfully.
Jul 26 14:46:12 cvm-perf13 cvmfs2[1531918]: (cvmfs-config.cern.ch) CernVM-FS: unmounted /cvmfs/cvmfs-config.cern.ch (cvmfs-config.cern.ch)
Jul 26 14:46:13 cvm-perf13 automount[1531841]: umount_autofs_indirect: ask umount returned busy /cvmfs
Jul 26 14:46:14 cvm-perf13 systemd[1]: autofs.service: Deactivated successfully.
Jul 26 14:46:14 cvm-perf13 systemd[1]: autofs.service: Unit process 1531909 (cvmfs2) remains running after unit stopped.
Jul 26 14:46:14 cvm-perf13 systemd[1]: autofs.service: Unit process 1531913 (cvmfs2) remains running after unit stopped.
Jul 26 14:46:14 cvm-perf13 systemd[1]: autofs.service: Unit process 1531975 (cvmfs2) remains running after unit stopped.
Jul 26 14:46:14 cvm-perf13 systemd[1]: autofs.service: Unit process 1531979 (cvmfs2) remains running after unit stopped.
Jul 26 14:46:14 cvm-perf13 systemd[1]: Stopped Automounts filesystems on demand.
Jul 26 14:46:14 cvm-perf13 systemd[1]: Starting Automounts filesystems on demand...
Jul 26 14:46:14 cvm-perf13 systemd[1]: tmp-autoR8DdD8.mount: Deactivated successfully.
Jul 26 14:46:14 cvm-perf13 systemd[1]: Started Automounts filesystems on demand.
Jul 26 14:46:14 cvm-perf13 systemd[1]: Started /usr/bin/systemctl start man-db-cache-update.
Jul 26 14:46:15 cvm-perf13 systemd[1]: tmp-autodJ2gDX.mount: Deactivated successfully.
Jul 26 14:46:24 cvm-perf13 systemd[1]: Reloading Automounts filesystems on demand...
Jul 26 14:46:24 cvm-perf13 systemd[1]: Reloaded Automounts filesystems on demand.
Jul 26 14:46:24 cvm-perf13 kernel: automount[1553460]: segfault at 7f41b32e7510 ip 00007f41b32e7510 sp 00007f41a3ffec98 error 14 in lookup_program.so[7f41b32f4000+2000] likely on CPU 20 (core 8, socket 1)
Jul 26 14:46:24 cvm-perf13 kernel: Code: Unable to access opcode bytes at 0x7f41b32e74e6.
Jul 26 14:46:24 cvm-perf13 systemd[1]: Started Process Core Dump (PID 1553505/UID 0).
```

**Comment by vvolkl** [comment]
I can only reproduce this with **`sssd-client-2.7.3-4.el9_1.3`** installed - it seems this is already fixed in **`ssd-client-2.8.2-2.el9`**

**Comment by jblomer** [comment]
Excellent! Many thanks for the investigation!

Then let's close this.
